src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/validsendtointerloc/ValidSendToInterlocMidEventHandler.java

```diff
@@ -0,0 +1,88 @@
+package uk.gov.hmcts.reform.sscs.ccd.presubmit.validsendtointerloc;
+
+import static java.util.Objects.requireNonNull;
+import static org.apache.commons.lang3.StringUtils.isBlank;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.sscs.ccd.callback.Callback;
+import uk.gov.hmcts.reform.sscs.ccd.callback.CallbackType;
+import uk.gov.hmcts.reform.sscs.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.sscs.ccd.domain.*;
+import uk.gov.hmcts.reform.sscs.ccd.presubmit.PreSubmitCallbackHandler;
+import uk.gov.hmcts.reform.sscs.docassembly.GenerateFile;
+import uk.gov.hmcts.reform.sscs.model.docassembly.GenerateFileParams;
+import uk.gov.hmcts.reform.sscs.model.docassembly.PostponeRequestTemplateBody;
+
+@Service
+public class ValidSendToInterlocMidEventHandler implements PreSubmitCallbackHandler<SscsCaseData> {
+
+    private static final String TITLE = "Postponement Request";
+    static final String FILENAME = "Postponement Request.pdf";
+    private final GenerateFile generateFile;
+    private final String templateId;
+
+    public ValidSendToInterlocMidEventHandler(GenerateFile generateFile, @Value("${doc_assembly.postponementrequest}") String templateId) {
+        this.generateFile = generateFile;
+        this.templateId = templateId;
+    }
+
+    @Override
+    public boolean canHandle(CallbackType callbackType, Callback<SscsCaseData> callback) {
+        requireNonNull(callback, "callback must not be null");
+        requireNonNull(callbackType, "callbacktype must not be null");
+
+        return callbackType.equals(CallbackType.MID_EVENT)
+                && (callback.getEvent() == EventType.VALID_SEND_TO_INTERLOC
+                || callback.getEvent() == EventType.ADMIN_SEND_TO_INTERLOCUTORY_REVIEW_STATE);
+    }
+
+    @Override
+    public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType, Callback<SscsCaseData> callback, String userAuthorisation) {
+        final SscsCaseData sscsCaseData = callback.getCaseDetails().getCaseData();
+        final PreSubmitCallbackResponse<SscsCaseData> response = new PreSubmitCallbackResponse<>(sscsCaseData);
+
+        final String requestDetails = sscsCaseData.getPostponementRequest().getPostponementRequestDetails();
```

Comment:
What happens if postponement request is blank here, bearing in mind this is a live event? Will they always get an error?

Reply:
Is it not mandatory on exui?

Reply:
Fixed